datafusion/expr/src/interval_arithmetic.rs

```diff
@@ -332,6 +332,46 @@ impl Interval {
         Ok(Self::new(unbounded_endpoint.clone(), unbounded_endpoint))
     }
 
+    /// Creates an interval between -∞ to ∞.
+    pub fn make_infinity_interval(data_type: &DataType) -> Result<Self> {
```

Comment:
We use `ScalarValue::Float(None)` for infinite bounds by convention. Even if you give Float(Some(f::INF)) or Float(Some(f::NaN)), they are converted to Float(None)'s during interval creation. We had given such a decision to have unique representation of unboundedness. You can check the details here: https://github.com/apache/datafusion/blob/deef834e7adfd859414448ab4da461e2d4eabb9e/datafusion/expr/src/interval_arithmetic.rs#L178

I recommend here to use `make_unbounded` API with floating types to not break this convention.